api/filter/http_connection_manager.proto

```diff
@@ -103,27 +103,56 @@ message AccessLogFilter {
   }
 }
 
+// Extra headers to include when doing AccessLogs via proto format.
+message HeaderSelection {
+  // Request headers to include in the AccessLog.
+  repeated string request_headers = 1;
+  // Response headers to include in the AccessLog.
+  repeated string response_headers = 2;
+}
+
 message AccessLog {
-  // Path the access log is written to.
-  // TODO(dpn): Should this be a one_of and allow a remote log gRPC endpoint
-  // in conjunction with the proto access log format?
-  string path = 1;
-
-  // Access log format. Envoy supports custom access log formats as well as a
-  // default format.
-  // TODO(dpn): This needs to be given some proto structure to allow control
-  // over what gets included
-  // [V2-API-DIFF] In addition to the existing format specifiers, the following
-  // additional specifiers will be available:
-  // %TLS_SNI_HOSTNAME%: SNI from handshake.
-  // %TLS_VERSION%: Possible values include: “TLSv1”, “TLSv1.1”,
-  //     “TLSv1.2”, “TLSv1.3”.
-  // %TLS_CIPHER_SUITE%: Cipher suite negotiated during the TLS handshake. The
-  //     value is four hex digits defined by the IANA TLS Cipher Suite Registry,
-  //     e.g. “009C” for TLS_RSA_WITH_AES_128_GCM_SHA256.
-  // TODO(htuch): Document how envoy.http_connection_manager.access_log values
-  // can be accessed in the format specifier.
-  string format = 2;
+  // Destination location the access log is written to.
+  //
+  // Envoy supports logging to both RPC endpoints and local disk files.
+  // Behavior is set by choosing which of these is used.
+  oneof destination {
+    // A path to a local file
+    string path = 1;
+    // An RPC endpoint
+    string endpoint = 5;
```

Comment:
SGTM.  I intend to have a statically registered log delivery modules to handle the need to do Stackdriver compatible logging but also give the ability for someone else to wrap the EnvoyAccessLog proto in their own custom way or ship it directly.  In both cases we can use the cluster name string as the endpoint, and the statically registered module takes care of any last chance wrapping or formatting required.